src/common/WebLink.js

```diff
@@ -1,48 +1,56 @@
 /* @flow strict-local */
 
-import React, { PureComponent } from 'react';
+import React from 'react';
 
-import type { Dispatch } from '../types';
-import { connect } from '../react-redux';
+import invariant from 'invariant';
+import { useSelector } from '../react-redux';
 import Label from './Label';
 import openLink from '../utils/openLink';
-import { getCurrentRealm } from '../selectors';
+import { tryGetCurrentRealm } from '../selectors';
 import { BRAND_COLOR, createStyleSheet } from '../styles';
 
 type Props = $ReadOnly<{|
-  dispatch: Dispatch,
   label: string,
   href: string,
-  realm: URL,
+
+  // What to use to absolutize relative URLs, if different from the
+  // current realm or we don't expect there to be a current realm.
+  baseUrl?: URL,
```

Comment:
This feels a bit fragile, in that even if a `baseUrl` is passed, the behavior will depend on whether a current realm is present. As we saw, that fact about a call site can change without there being any obvious reason to look through to the details of this component.

How about we do away with this component's looking at the current realm entirely, and have it always take `baseUrl` from the caller? And if that's missing, it just passes `undefined` through to `new URL`, requiring the href itself to be absolute.

Reply:
(There's currently just the one callsite, so this doesn't involve any major refactor.)

Reply:
> How about we do away with this component's looking at the current realm entirely, and have it always take `baseUrl` from the caller? And if that's missing, it just passes `undefined` through to `new URL`, requiring the href itself to be absolute.

Hmm, actually, what do you think of just asking all callers to provide the entire absolute URL as a URL object, so the validation happens outside `WebLink`? If we're going to always require (a) an absolute URL, or (b) a base URL plus a relative URL (which together make an absolute URL), we might as well make it simpler and always require an absolute URL?

Reply:
Sure, SGTM.